src/main/host/descriptor/udp.c

```diff
@@ -212,9 +238,20 @@ static gboolean _udp_close(Descriptor* descriptor) {
     UDP* udp = _udp_fromDescriptor(descriptor);
     MAGIC_ASSERT(udp);
     /* Deregister us from the process upon return. */
+    _udp_setState(udp, UDPS_CLOSED);
     return TRUE;
 }
 
+gint udp_shutdown(UDP* udp, gint how) {
```

Comment:
Why passing the `how` argument here ? This argument looks to indicate how to close the file descriptor (`how != SHUT_RD && how != SHUT_WR && how != SHUT_RDWR`) but I have never heard about a way to close file descriptor depending on the way it was opened... Could you please explain me how it works?

Reply:
For UDP sockets, we aren't currently using the `how` argument, but we should be if we wanted our UDP tests to pass in Shadow. This `udp_shutdown()` function should imitate the `int shutdown(int sockfd, int how)` [system call](https://man7.org/linux/man-pages/man2/shutdown.2.html), which allows you to separately shut down the read or write ends of a socket. For TCP, calling `shutdown(fd, SHUT_WR)` would send a `FIN` packet and prevent you from writing any more to the socket, but still allowing you to continue reading on the socket. For UDP this `shutdown()` call is much less meaningful and I think should just set a flag to prevent the program from either reading or writing new messages.